connector/src/main/java/org/geysermc/connector/entity/Entity.java

```diff
@@ -105,6 +105,9 @@ public Entity(long entityId, long geyserId, EntityType entityType, Vector3f posi
     public void spawnEntity(GeyserSession session) {
         AddEntityPacket addEntityPacket = new AddEntityPacket();
         addEntityPacket.setIdentifier("minecraft:" + entityType.name().toLowerCase());
+        if (entityType.name().toLowerCase().equals("tropical_fish")) {
```

Comment:
Just override the `spawnEntity` method in the tropical/abstract fish entity class.

Reply:
See the villager entity class for an example.